File: csharp/CodexFeatures/MultiFileContext.cs
using System.IO;
using System.Text;

namespace TormentNexus.CodexFeatures
{
    public class MultiFileContext
    {
        public string GatherContext(string directory, int maxFiles = 10)
        {
            var context = new StringBuilder();
            int fileCount = 0;

            if (Directory.Exists(directory))
            {
                var files = Directory.GetFiles(directory);
                foreach (var file in files)
                {
                    if (fileCount >= maxFiles) break;

                    var fileName = Path.GetFileName(file);
                    if (!fileName.StartsWith("."))
                    {
                        context.AppendLine($"\n--- File: {fileName} ---");
                        var content = File.ReadAllText(file);
                        context.Append(content.Length > 1000 ? content.Substring(0, 1000) : content);
                        fileCount++;
                    }
                }
            }

            return context.ToString();
        }
    }
}

File: java/src/main/java/com/tormentnexus/codex/MultiFileContext.java
package com.tormentnexus.codex;

import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;

public class MultiFileContext {
    public String gatherContext(String directory, int maxFiles) {
        StringBuilder context = new StringBuilder();
        int fileCount = 0;

        File dir = new File(directory);
        File[] files = dir.listFiles();

        if (files != null) {
            for (File file : files) {
                if (fileCount >= maxFiles) break;

                if (file.isFile() && !file.getName().startsWith(".")) {
                    context.append("\n--- File: ").append(file.getName()).append(" ---\n");
                    try {
                        String content = new String(Files.readAllBytes(Paths.get(file.getAbsolutePath())));
                        context.append(content.length() > 1000 ? content.substring(0, 1000) : content);
                        fileCount++;
                    } catch (IOException e) {
                        // ignore unreadable files
                    }
                }
            }
        }
        return context.toString();
    }
}